spectrocloud/resource_application_profile.go

```diff
@@ -0,0 +1,546 @@
+		UpdateContext: resourceApplicationProfileUpdate,
+		DeleteContext: resourceApplicationProfileDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Second),
+			Update: schema.DefaultTimeout(20 * time.Second),
+			Delete: schema.DefaultTimeout(20 * time.Second),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "1.0.0", // default as in UI
+			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "project",
+				ValidateFunc: validation.StringInSlice([]string{"", "project", "tenant", "system"}, false),
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"cloud": {
+				Type:     schema.TypeString,
+				Default:  "all",
+				Optional: true,
+				ForceNew: true,
+			},
+			"pack": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "spectro",
+						},
+						"source_app_tier": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"registry_uid": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+						"uid": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"manifest": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"uid": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"content": {
+										Type:     schema.TypeString,
+										Required: true,
+										DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+											// UI strips the trailing newline on save
+											if strings.TrimSpace(old) == strings.TrimSpace(new) {
+												return true
+											}
+											return false
+										},
+									},
+								},
+							},
+						},
+						//"layer": {
+						//	Type:     schema.TypeString,
+						//	Required: true,
+						//},
+						"tag": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"values": {
+							Type:     schema.TypeString,
+							Optional: true,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								// UI strips the trailing newline on save
```

Comment:
```go
return strings.TrimSpace(old) == strings.TrimSpace(new)
```

Reply:
Changed to your suggestion for readability.